src/Components/SysTable/SysTable.js

```diff
@@ -1,183 +1,173 @@
-        return chips;
-    };
-
-    const activeFiltersConfig = {
-        deleteTitle: intl.formatMessage(messages.resetFilters),
-        filters: buildFilterChips(),
-        onDelete: (event, itemsToRemove, isAll) => {
-            if (isAll) {
-                sysTableFilters({});
-                stateSet({ type: 'setTableVars', payload: { name: '' } });
-            } else {
-                itemsToRemove.map((item) => {
-                    item.value === 'name' && stateSet({ type: 'setTableVars', payload: { name: '' } });
-                    item.value === 'matched' && sysTableFilters({});
-                });
-            }
-        }
-    };
-
-    useEffect(() => {
-        const rowBuilder = data => data?.hostsList?.flatMap((data, key) => ({
-            rowId: key,
-            cells: [
-                { title: <Link to={`/systems/${data.displayName}`}>{data.displayName}</Link> },
-                { title: <StatusLabel isDisabled={data.isDisabled} hasMatch={data.hasMatch} displayMatch /> },
-                {
-                    title: data.lastMatchDate ?
-                        <Tooltip content={<DateFormat date={new Date(data.lastMatchDate)} type='exact' />}>
-                            <span><DateFormat date={new Date(data.lastMatchDate)} type='onlyDate' /></span>
-                        </Tooltip>
-                        : <Tooltip content={intl.formatMessage(messages.noHostHas)}>
-                            <span>{intl.formatMessage(messages.never)}</span>
-                        </Tooltip>
-                },
-                { title: <Link to={`/systems/${data.displayName}`}>{data.totalMatches?.toLocaleString()}</Link> },
-                { title: <Tooltip content={<DateFormat date={new Date(data.lastScanDate)} type='exact' />}>
-                    <span><DateFormat date={new Date(data.lastScanDate)} /></span>
-                </Tooltip> }
-
-            ]
-        }));
-
-        stateSet({ type: 'setRows', payload: rowBuilder(sysTableData) });
-    }, [intl, sysTableData]);
+    const fetchSystems = useFetchSystems({
+        query: GET_SYSTEM_TABLE
+    });
+
+    const mergedColumns = (defaultColumns) =>
+        columns.map((column) => {
+            const isStringCol = typeof column === 'string';
+            const key = isStringCol ? column : column.key;
+            const defaultColumn = defaultColumns.find(
+                (defaultCol) => defaultCol.key === key
+            );
+            return {
+                ...defaultColumn,
+                ...(isStringCol ? { key: column } : column),
+                props: {
+                    ...defaultColumn?.props,
+                    ...column?.props
+                }
+            };
+        });
+
+    const appendDirection = (attributes, direction) =>
+        attributes.map((attribute) => `${attribute}_${direction}`);
+
+    const findColumnByKey = (key) =>
```

Comment:
Same for these functions, could be worth moving outside of the component render cycle and into `utils` file.